### PR TITLE
Rename custom panels to dashboards

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
Rename panles to dashboards.

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]
how do users distinguish between this and core dashboards? the product opensearch dashboards and the dashboards plugin in it already caused confusions, not sure why we want one more dashboards..

File: dashboards-observability/public/components/application_analytics/home.tsx [hunk 1]
```diff
@@ -160,7 +161,7 @@ export const Home = (props: HomeProps) => {
     return http
       .post(`${CUSTOM_PANELS_API_PREFIX}/panels`, {
         body: JSON.stringify({
-          panelName: `${appName}'s Panel`,
+          panelName: `${appName}'s Dashboards`,
```
Comment (joshuali925) [inline dashboards-observability/public/components/application_analytics/home.tsx:164]: if panel is singular should it be dashboard instead of dashboards?

Comment (mengweieric) [reply dashboards-observability/public/components/application_analytics/home.tsx:164]: yes i think i'll need to use dashboard here

**Review by ps48 — APPROVED** [review]

**Comment by ps48** [comment]
@mengweieric  In my opinion, this is adding to more confusion already in place:
1. OpenSearch Dashboards
2. Dashboards plugin in OpenSearch Dashboards
3. Now this rename of Operational Panels to Dashboards in Observability

File: dashboards-observability/.cypress/integration/3_panels.spec.js [hunk 2]
```diff
@@ -97,25 +97,25 @@ describe('Creating visualizations', () => {
...
-describe('Testing panels table', () => {
+describe('Testing dashboards table', () => {
   beforeEach(() => {
     moveToPanelHome();
   });
 
-  it('Displays error toast for invalid panel name', () => {
-    cy.get('.euiButton__text').contains('Create panel').trigger('mouseover').click();
+  it('Displays error toast for invalid dashboards name', () => {
+    cy.get('.euiButton__text').contains('Create dashboards').trigger('mouseover').click();
```
[the remark this answers is not in the document]
Comment (mengweieric) [reply dashboards-observability/.cypress/integration/3_panels.spec.js:106]: Yes I'll later check and change all places

[the remark this answers is not in the document]
Comment (anirudha) [reply dashboards-observability/.cypress/integration/3_panels.spec.js:106]: Why are we changing this?

**Comment by mengweieric** [comment]
Paused, and not going into 2.4, close for now.

**Pull request closed without merging**